Source/MQTTnet/Client/MqttClient.cs

```diff
@@ -751,10 +755,22 @@ async Task<MqttClientPublishResult> PublishAtLeastOnce(MqttPublishPacket publish
 
         async Task<MqttClientPublishResult> PublishAtMostOnce(MqttPublishPacket publishPacket, CancellationToken cancellationToken)
         {
-            // No packet identifier is used for QoS 0 [3.3.2.2 Packet Identifier]
-            await Send(publishPacket, cancellationToken).ConfigureAwait(false);
+            try
+            {
+                // No packet identifier is used for QoS 0 [3.3.2.2 Packet Identifier]
+                await Send(publishPacket, cancellationToken).ConfigureAwait(false);
+
+                return MqttClientResultFactory.PublishResult.Create(null);
+            }
+            catch (Exception)
+            {
+                if (_unexpectedDisconnectPacket != null)
```

Comment:
Does this only apply for QoS 0? I am wondering if this code must be also used for QoS 1 and 2?

Reply:
I think it only applies to QoS 0 because for QoS 1, the client actually registers an awaitable and wait for the response packet, so it already throws an exception with the error code as of today. But for QoS 0, the client does not register any awaitable.  I guess this is by design because QoS 0 should not expect any response from server. But QoS 1 can always expect some response, whether its a PUBACK or DISCONNECT